homeassistant/components/discord/notify.py

```diff
@@ -25,30 +29,36 @@
 ATTR_EMBED_THUMBNAIL = "thumbnail"
 ATTR_IMAGES = "images"
 
+# Deprecated in Home Assistant 2022.4
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({vol.Required(CONF_TOKEN): cv.string})
 
 
-def get_service(hass, config, discovery_info=None):
-    """Get the Discord notification service."""
-    token = config[CONF_TOKEN]
-    return DiscordNotificationService(hass, token)
+async def async_get_service(
+    hass: HomeAssistant,
+    config: ConfigType,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> DiscordNotificationService:
+    """Get the NFAndroidTV notification service."""
+    if discovery_info is not None:
+        return DiscordNotificationService(hass, discovery_info[CONF_TOKEN])
+    return DiscordNotificationService(hass, config[CONF_TOKEN])
```

Comment:
When does this happen? Discovery info is now always there, right?

Reply:
Discovery info seems to be `None` when yaml config is still present for this `Domain`.

Reply:
Sure by YAML configuration should not be loaded?
It should be imported and only the config entry should be loaded.